src/components/QuoteBubble/QuoteBubble.js

```diff
@@ -1,40 +1,72 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-
-import extractRoleFromPath from '../../utils/extractRoleFromPath';
-import extractActionFromPath from '../../utils/extractActionFromPath';
-import extractQuoteIDFromPath from '../../utils/extractQuoteIDFromPath';
-import { getQuote } from '../../utils/quotesService';
-import { match, speechBubbleVariant } from '../../types';
-import { SPEECH } from '../../constants/speechBubbleVariant';
+import React, { useState } from 'react';
+import { useTransition } from 'react-spring';
+import { withRouter } from 'react-router-dom';
+import Measure from 'react-measure';
+
+import { location } from '../../types';
+import { THOUGHT, SPEECH } from '../../constants/speechBubbleVariant';
+import { colorWhite } from '../../styles/designTokens';
+import useBubble from './useBubble';
 import CopyButton from './CopyButton';
 import Wrapper from './Wrapper';
+import Bubble from './Bubble';
 import Quote from './Quote';
 import BubbleTail from './BubbleTail';
 
-function QuoteBubble({ variant = SPEECH, noCopyToClipboard = false, match }) {
-  const { url } = match;
+function QuoteBubble({ location }) {
+  const { pathname } = location;
+  const bubble = useBubble(pathname);
+
+  // Increases wrapper height for big quote bubbles
+  // wrapperHeight and Measure is pretty much required only for small mobile devices
+  // TODO check performance implications and maybe better way to handle it @blurbyte
+  const [wrapperHeight, setWrapperHeight] = useState(0);
 
-  const role = extractRoleFromPath(url);
-  const action = extractActionFromPath(url);
-  const quoteID = extractQuoteIDFromPath(url);
+  const transitions = useTransition(bubble, bubble => bubble.quoteID, {
+    from: { opacity: 0, transform: 'perspective(600px) rotateX(45deg) translateY(-20px) scale(0.8)' },
+    enter: { opacity: 1, transform: 'perspective(600px) rotateX(0deg) translateY(0) scaleY(1)' },
+    leave: {
+      opacity: 0,
+      transform: 'perspective(600px) rotateX(0deg) translateY(-30px) scale(0.6)',
+      color: colorWhite
+    },
+    config: {
+      mass: 1,
+      tension: 200,
+      friction: 14
+    },
+    delay: 200
+  });
 
-  const quote = getQuote(role, action, quoteID);
+  const handleResize = contentRect => {
+    const offsetHeight = contentRect.offset.height;
+    if (wrapperHeight !== offsetHeight) {
+      setWrapperHeight(offsetHeight);
+    }
+  };
 
   return (
-    <Wrapper>
-      <Quote>{quote}</Quote>
-      <BubbleTail variant={variant} />
-      {!noCopyToClipboard && <CopyButton valueToCopy={quote} />}
-    </Wrapper>
+    <Measure offset onResize={handleResize}>
```

Comment:
1. I would recommend using debounce. Since we already use `lodash` lib, we can use `debounce` method from it.
2. `QuoteBubble` component gets into _inifinite_ loop. (add console.log to the inside `handleResize` or in any place)


![leak](https://user-images.githubusercontent.com/14184100/61383572-3bf27780-a8af-11e9-9c34-c0ac7bd7081d.gif)

Reply:
Yeah, spotted same behaviour, therefore comment about perf optimisation.

Reply:
For now removing it, will try later to do it with **react-spring** alone.